dbms/src/Flash/Planner/Plans/PhysicalJoin.cpp

```diff
@@ -68,11 +68,8 @@ PhysicalPlanNodePtr PhysicalJoin::build(
     RUNTIME_CHECK(left);
     RUNTIME_CHECK(right);
 
-    left->finalize();
-    right->finalize();
-
-    const Block & left_input_header = left->getSampleBlock();
-    const Block & right_input_header = right->getSampleBlock();
+    const Block & left_input_header = Block(left->getSchema());
+    const Block & right_input_header = Block(right->getSchema());
```

Comment:
Do `probe/build_side_header` also need to be changed to `Block(probe/build_plan->getSchema())`?